setup.py

```diff
@@ -63,6 +63,7 @@
         "openpyxl>=3.0.0,!=3.1.0",
         "validators>=0.20.0",
         "matplotlib>=3.4.0",
+        "scikit-learn",
```

Comment:
let's not have this as a hard dependency of the package and instead do it as a lazy try/except import inside only the tlm_calibrated file / functions where sklearn is needed

Reply:
in the future would be good to call out in PR description if you are introducing a new dependency to the client, that is a major change